test/unit/carousel.ts

```diff
@@ -0,0 +1,403 @@
+  let carousel: Carousel;
+  let DEFAULT_SETTINGS;
+
+  describe('settings', () => {
+    it('should default to default settings if no settings are passed in', () => {
+      carousel = new Carousel();
+      const settings = {
+        speed: 0,
+        slidesToShow: 1,
+        slidesToScroll: 1
+      };
+
+      expect(carousel.state.settings).to.deep.eq(settings);
+    });
+  });
+
+  describe('functions', () => {
+    beforeEach(() => {
+      carousel = new Carousel();
+      carousel.props.items = [
+        { content: 'test0' },
+        { content: 'test1' },
+        { content: 'test2' },
+        { content: 'test3' },
+        { content: 'test4' },
+        { content: 'test5' },
+        { content: 'test6' },
+        { content: 'test7' },
+        { content: 'test8' },
+        { content: 'test9' },
+      ];
+    });
+
+    describe('init()', () => {
+      it('should initialize state settings', () => {
+        carousel.props.settings = { speed: 500 };
+        carousel.init();
+        expect(carousel.state.settings).to.deep.eq({ speed: 500, slidesToScroll: 1, slidesToShow: 1 });
+      });
+    });
+
+    describe('onMount()', () => {
+      it('should add event listener for window resize', () => {
+        const addEventListener = spy();
+        stub(utils, 'WINDOW').returns({ addEventListener });
+        carousel.onMount();
+        expect(addEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('onUpdate()', () => {
+      it('should add cloned items to state', () => {
+        const clone = carousel.cloneItems = spy();
+        carousel.onUpdate();
+        expect(clone).to.be.called;
+      });
+    });
+
+    describe('onUnMount()', () => {
+      it('should add event listener for window resize', () => {
+        const removeEventListener = spy();
+        stub(utils, 'WINDOW').returns({ removeEventListener });
+
+        carousel.onUnMount();
+
+        expect(removeEventListener).to.be.calledWithExactly('resize', carousel.update);
+      });
+    });
+
+    describe('moveNext', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.moveNext();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(1);
+      });
+    });
+
+    describe('movePrevious', () => {
+      it('should call slideHandler function with correct slide number', () => {
+        carousel.state.currentSlide = 0;
+        carousel.slideHandler = spy();
+
+        carousel.movePrevious();
+
+        expect(carousel.slideHandler).to.be.calledWithExactly(-1);
+      });
+    });
+
+    describe('isSwipeToNext()', () => {
+      it('should move next when swiping left with an angle between 315 and 360', () => {
+        const direction = Carousel.isSwipeToNext({ startX: 0, startY: 0 }, -25, 4);
+        expect(direction).to.eq(true);
```

Comment:
`.to.be.true` and `.to.be.false`. Only going to leave one comment to avoid spam.